### PR TITLE
New module azure_rm_managementgroup_info

### PR DESCRIPTION
**Pull request by paultaiton**
##### SUMMARY

info module for management groups.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

azure_rm_managementgroup_info

##### ADDITIONAL INFORMATION

This is a read only info module for querying Azure RM and gather details about management groups.

I have added integration test tasks, but I'm not sure what kind of role assignments exist for the test accounts, and can't be 100% certain they will work. It requires at least 1 management group exists which the principal has reader access to.

### CODE REVIEW

File: plugins/modules/azure_rm_managementgroup_info.py [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+
+
+class AzureRMManagementGroupInfo(AzureRMModuleBase):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            children=dict(type='bool', default=False),
+            flatten=dict(type='bool', default=False),
+            id=dict(type='str'),
+            name=dict(type='str'),
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_managementgroup_info.py]:
```suggestion
            name=dict(type='str'， aliases=['management_group_name']),
```

Comment (Fred-sun) [reply plugins/modules/azure_rm_managementgroup_info.py]: Please add aliases info to arg define.

**Review by Fred-sun — COMMENTED** [review]
Please add follow lines to ~/tests/sanity/ignore-2.10.txt and ~/tests/sanity/ignore-2.11.txt
```e.g.
plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-requirements-unknown
plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-unknown-key

File: plugins/modules/azure_rm_managementgroup_info.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+# Copyright (c) 2021 Paul Aiton, < @paultaiton >
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from msrestazure.tools import parse_resource_id
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_managementgroup_info.py]:
```suggestion

```

File: plugins/modules/azure_rm_managementgroup_info.py [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+                    'status': ['preview'],
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_managementgroup_info.py]:
```suggestion

```

File: plugins/modules/azure_rm_managementgroup_info.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+                    'supported_by': 'community'}
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_managementgroup_info.py]:
```suggestion

```

**Review by Fred-sun — COMMENTED** [review]
please remove the ANSIBLE_METADATA block, it is no longer needed.

File: plugins/modules/azure_rm_managementgroup_info.py [hunk 5]
```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2021 Paul Aiton, < @paultaiton >
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+from msrestazure.tools import parse_resource_id
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_managementgroup_info.py]:
```suggestion

```

File: tests/integration/targets/azure_rm_managementgroup/aliases [hunk 6]
```diff
@@ -0,0 +1,3 @@
+cloud/azure
+shippable/azure/group2
+destructive
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_managementgroup/aliases]:
```suggestion
disabled
destructive
```

**Review by Fred-sun — COMMENTED** [review]
Please add "disabled" to tests/integration/targets/azure_rm_managementgroup/aliases, Thank you very mcuh!

**Comment by paultaiton** [comment]
Hello,

Your suggestions have been committed. Thank you for the review @Fred-sun

File: tests/sanity/ignore-2.10.txt [hunk 7]
```diff
@@ -206,6 +206,8 @@ plugins/modules/azure_rm_manageddisk_info.py validate-modules:return-syntax-erro
 plugins/modules/azure_rm_managementgroup.py validate-modules:invalid-ansiblemodule-schema
 plugins/modules/azure_rm_managementgroup.py validate-modules:required_if-requirements-unknown
 plugins/modules/azure_rm_managementgroup.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-requirements-unknown
```
Comment (Fred-sun) [inline tests/sanity/ignore-2.10.txt:209]:
```suggestion

```

File: tests/sanity/ignore-2.10.txt [hunk 8]
```diff
@@ -206,6 +206,8 @@ plugins/modules/azure_rm_manageddisk_info.py validate-modules:return-syntax-erro
...
+plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-unknown-key
```
Comment (Fred-sun) [inline tests/sanity/ignore-2.10.txt:210]: unnecessary
```suggestion

```

File: plugins/modules/azure_rm_managementgroup_info.py [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common import AzureRMModuleBase
+
+
+class AzureRMManagementGroupInfo(AzureRMModuleBase):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            children=dict(type='bool', default=False),
+            flatten=dict(type='bool', default=False),
+            id=dict(type='str'),
+            name=dict(type='str'， aliases=['management_group_name']),
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_managementgroup_info.py]: invalid character in identifie
```suggestion
            name=dict(type='str', aliases=['management_group_name']),
```

File: plugins/modules/azure_rm_managementgroup_info.py [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+            else:
+                mg_name = self.name
+
+        expand = 'children' if self.children else None
+        try:
+            response = self.management_groups_client.management_groups.get(group_id=mg_name,
+                                                                           expand=expand,
+                                                                           recurse=self.recurse)
+        except CloudError:
+            self.log('No Management group {} found.'.format(mg_name))
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_managementgroup_info.py]:
```suggestion
            self.log('No Management group {0} found.'.format(mg_name))
```

File: tests/sanity/ignore-2.11.txt [hunk 11]
```diff
@@ -206,6 +206,8 @@ plugins/modules/azure_rm_manageddisk_info.py validate-modules:return-syntax-erro
 plugins/modules/azure_rm_managementgroup.py validate-modules:invalid-ansiblemodule-schema
 plugins/modules/azure_rm_managementgroup.py validate-modules:required_if-requirements-unknown
 plugins/modules/azure_rm_managementgroup.py validate-modules:required_if-unknown-key
+plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-requirements-unknown
+plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-unknown-keyp
```
Comment (Fred-sun) [inline tests/sanity/ignore-2.11.txt]:
```suggestion
plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-unknown-key
```

**Review by Fred-sun — COMMENTED** [review]
@paultaiton It's my mistake! Please add follow lines to ~/tests/sanity/ignore-2.10.txt
```e.g.
plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-requirements-unknown
plugins/modules/azure_rm_managementgroup_info.py validate-modules:required_if-unknown-key

**Comment by paultaiton** [comment]
> @paultaiton It's my mistake! Please add follow lines to ~/tests/sanity/ignore-2.10.txt

haha, I was wondering. No problem sir, we're all on the same team.

**Comment by Fred-sun** [comment]
@paultaiton There should be no problem. We will complete the review as soon as possible and recommend the merger! Thank you very much!

**Comment by paultaiton** [comment]
> @paultaiton There should be no problem. We will complete the review as soon as possible and recommend the merger! Thank you very much!

Awesome. Thank you very much @Fred-sun

File: plugins/module_utils/azure_rm_common.py [hunk 12]
```diff
@@ -233,6 +234,7 @@ def default_api_version(self):
     from azure.mgmt.web.version import VERSION as web_client_version
     from azure.mgmt.network import NetworkManagementClient
     from azure.mgmt.resource.resources import ResourceManagementClient
+    from azure.mgmt.managementgroups import ManagementGroupsAPI
```
Comment (Fred-sun) [inline plugins/module_utils/azure_rm_common.py]:
```suggestion
    from azure.mgmt.managementgroups import ManagementGroupsAPI as ManagementGroupsClient
```

**Review by Fred-sun — COMMENTED** [review]
Follow other module's define!

**Comment by paultaiton** [comment]
@Fred-sun 
I had to update a couple lines in addition to your suggestion to make it work, but they are now pushed into the branch.

**Comment by paultaiton** [comment]
Thanks @haiyuazhang , appreciate it.

**Pull request merged**